Fixes #646 - Investigate failure of test_that_checks_the_deletion_of_a_review and test_that_checks_the_editing_of_a_review

bobsilverberg opened the pull request:
Address issues causing tests to fail

Use a unique user per test and create a new review each time
Locate a review by a user by their username, rather than by the fact that there is a delete link
Rename ReviewSnippet to Review

This should stabilize these tests which have been having a lot of issues.

@m8ttyB / @rbillings r?

m8ttyB commented on pages/desktop/consumer_pages/reviews.py:
```diff
@@ -27,24 +27,29 @@ def __init__(self, testsetup, app_name=False):
         for review in self.reviews:
-            if review.find_element(*review._delete_review_locator):
+            if review.author == user_name:
                 break
```
Honestly might be due to #toddlerlife #nosleep effecting me, should `return review` be moved into this `if` block?

bobsilverberg replied:
You are absolutely right, @m8ttyB. This method could be greatly improved. I'll fix it tomorrow and ask for another review. Thanks!

m8ttyB commented on tests/desktop/consumer_pages/test_reviews.py:
```diff
@@ -16,36 +14,20 @@
         home_page.header.click_sign_in()
-        acct = self.create_new_user(mozwebqa)
-        home_page.login(acct)
+        self.acct = self.create_new_user(mozwebqa)
```
Can you create a code level comment as to why the use of `self.`?

m8ttyB replied:
Also, small question, why a class level `self.` here?

bobsilverberg commented:
@m8ttyB I fixed up the `get_review_for_user` method and added a comment about why we are storing the `acct` variable in `self`.

Unfortunately these tests won't currently pass on dev due to the new nav which has broken pretty much all of our tests. I hope to get a PR out for that breakage soon.

m8ttyB commented:
r+

m8ttyB commented:
merging
